datafusion/sql/src/relation/mod.rs

```diff
@@ -142,10 +143,36 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                 );
             }
         };
+
+        let optimized_plan = optimize_subquery_sort(plan)?.data;
         if let Some(alias) = alias {
-            self.apply_table_alias(plan, alias)
+            self.apply_table_alias(optimized_plan, alias)
         } else {
-            Ok(plan)
+            Ok(optimized_plan)
         }
     }
 }
+
+fn optimize_subquery_sort(plan: LogicalPlan) -> Result<Transformed<LogicalPlan>> {
+    // When a subquery is initialized, we look for the sort options since they might be redundant.
+    // It's only important if the subquery result is affected with the order by statement.
+    // Which are the cases of:
+    // 1. DISTINCT ON / ARRAY_AGG ... => It's handled as an Aggregate plan and keep the sorting
+    // 2. RANK / ROW_NUMBER ... => It's handled as a WindowAggr plan sorting is preserved
+    // 3. LIMIT => It's handled as a Sort plan type so that we need to search for it
```

Comment:
```suggestion
    // When a subquery is initialized, we look for the sort options since they might be unnecessary.
    // It's only important if the subquery result is affected by the ORDER BY statement, which can
    // happen when we have:
    // 1. DISTINCT ON / ARRAY_AGG ... => It's handled as an Aggregate plan and keep the sorting
    // 2. RANK / ROW_NUMBER ... => It's handled as a WindowAggr plan sorting is preserved
    // 3. LIMIT => It's handled as a Sort plan type so that we need to search for it
```